core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```
@@ -140,6 +141,71 @@ class KRaftMetadataCache(val brokerId: Int) extends MetadataCache with Logging w
     }
   }
 
+  private def getPartitionMetadataForDescribeTopicResponse(
+    image: MetadataImage,
+    topicName: String,
+    listenerName: ListenerName
+  ): Option[List[DescribeTopicPartitionsResponsePartition]] = {
+    Option(image.topics().getTopic(topicName)) match {
+      case None => None
+      case Some(topic) => {
+        val partitions = Some(topic.partitions().entrySet().asScala.map { entry =>
+          val partitionId = entry.getKey
+          val partition = entry.getValue
+          val filteredReplicas = maybeFilterAliveReplicas(image, partition.replicas,
+            listenerName, false)
+          val filteredIsr = maybeFilterAliveReplicas(image, partition.isr, listenerName,
+            false)
+          val offlineReplicas = getOfflineReplicas(image, partition, listenerName)
+          val maybeLeader = getAliveEndpoint(image, partition.leader, listenerName)
+          maybeLeader match {
+            case None =>
+              val error = if (!image.cluster().brokers.containsKey(partition.leader)) {
```

I'm not sure this should be an error for the describe topics results -- the describe succeeded, not having a leader is on of the states that the partition can be in.

---

Good question, actually I am not super sure about these errors. They may have a history as they are included in the getPartitionMetadata method. @mumrah Do you have an idea?

---

This won't cause an error for the whole request right? It will just populate the partition-level `ErrorCode`

---

I guess we need to see what the client does with the error code.

---

At least from the admin client perspective, when handling the MetadataRequest(where this error handling copied from), partition level errors are ignored. In our case, the DescribeTopicPartition request currently is only used by the admin client, so I will remove the errors.
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java#L163